packages/core/src/Step.js

```diff
@@ -0,0 +1,37 @@
+import React from 'react'
+import PropTypes from 'prop-types'
+import styled from 'styled-components'
+import Flex from './Flex'
+import Text from './Text'
+import Icon from 'pcln-icons'
+
+const Container = styled(Flex)`
+  cursor: ${props => (props.onClick ? 'pointer' : 'default')};
+`
+
+function Step({ className, isCurrent, isCompleted, children, onClick }) {
+  const color = isCurrent || isCompleted ? 'primary' : 'text.light'
+
+  return (
+    <Container className={className} alignItems="center" onClick={onClick}>
+      {isCompleted && (
+        <Icon name="check" color="primary" size={16} mr={1} legacy={false} />
+      )}
+      <Text color={color} bold={isCurrent} fontSize={1}>
+        {children}
+      </Text>
+    </Container>
+  )
+}
+
+Step.displayName = 'Step'
+
+Step.propTypes = {
+  className: PropTypes.string,
+  isCurrent: PropTypes.bool,
+  isCompleted: PropTypes.bool,
+  children: PropTypes.node,
```

Comment:
Prop type for children can use further prop type definite with `PropTypes.element`. (See [prop-types usage doc](https://github.com/facebook/prop-types#usage) to learn more.)

Reply:
I prefer `node` here as the children could be a string, highly unlikely but I don't think we should through a warning if that occurs.

Reply:
It’s fair to render string as children without warning. Good point. This comment could be resolved. Thanks.